Emby.Server.Implementations/Syncplay/SyncplayController.cs

```diff
@@ -228,190 +271,271 @@ public void SessionJoin(SessionInfo session, JoinGroupRequest request)
-
-                            _group.LastActivity = currentTime.AddMilliseconds(
-                                delay
-                            );
-
-                            var command = NewSyncplayCommand(SendCommandType.Play);
-                            SendCommand(session, BroadcastType.AllGroup, command);
-                        }
-                    }                    
-                }
-                else
-                {
-                    // Make sure client has latest group state
-                    var command = NewSyncplayCommand(SendCommandType.Play);
-                    SendCommand(session, BroadcastType.SingleSession, command);
+                        // Client, that was buffering, resumed playback but did not update others in time
+                        delay = _group.GetHighestPing() * 2;
+                        delay = delay < _group.DefaulPing ? _group.DefaulPing : delay;
+
+                        _group.LastActivity = currentTime.AddMilliseconds(
+                            delay
+                        );
+
+                        var command = NewSyncplayCommand(SendCommandType.Play);
+                        SendCommand(session, BroadcastType.AllGroup, command, CancellationToken.None);
+                    }
                 }
             }
-            else if (request.Type.Equals(PlaybackRequestType.UpdatePing))
+            else
             {
-                _group.UpdatePing(session, request.Ping ??= _group.DefaulPing);
+                // Group was not waiting, make sure client has latest state
+                var command = NewSyncplayCommand(SendCommandType.Play);
+                SendCommand(session, BroadcastType.CurrentSession, command, CancellationToken.None);
             }
         }
 
+        /// <summary>
+        /// Sanitizes the PositionTicks, considers the current playing item when available.
+        /// </summary>
+        /// <param name="positionTicks">The PositionTicks.</param>
+        /// <value>The sanitized PositionTicks.</value>
+        private long SanitizePositionTicks(long? positionTicks)
+        {
+            var ticks = positionTicks ??= 0;
+            ticks = ticks >= 0 ? ticks : 0;
+            if (_group.PlayingItem.RunTimeTicks != null)
+            {
+                var runTimeTicks = _group.PlayingItem.RunTimeTicks ??= 0;
+                ticks = ticks > runTimeTicks ? runTimeTicks : ticks;
+            }
+            return ticks;
+        }
+
+        /// <summary>
+        /// Updates ping of a session.
+        /// </summary>
+        /// <param name="session">The session.</param>
+        /// <param name="request">The update.</param>
+        private void HandlePingUpdateRequest(SessionInfo session, PlaybackRequest request)
+        {
+            // Collected pings are used to account for network latency when unpausing playback
+            var ping = request.Ping ??= _group.DefaulPing;
```

Comment:
Why are you using `??=` instead of `??` everywhere? AFAIK it updates the field if it was `null`, why do we want to update `request` during its handling? I would assume request should be considered read-only...

Reply:
In my quick google-search I misread what `??=` does. `??` is indeed what I want in the majority of the cases, I'll review them all.